### PR TITLE
[SPARK-46208][PS][DOCS] Adding a link for latest Pandas API specifications.

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to adding a link for latest Pandas API specifications.

### Why are the changes needed?

To increase the usability of documents

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The existing CI should pass

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/docs/source/reference/pyspark.pandas/index.rst [hunk 1]
```diff
@@ -23,7 +23,7 @@ Pandas API on Spark
 This page gives an overview of all public pandas API on Spark.
 
 .. note::
-   pandas API on Spark follows the API specifications of latest pandas release.
```
Comment (itholic) [inline python/docs/source/reference/pyspark.pandas/index.rst:26]: I believe we cannot always support the latest pandas version in some cases, especially when they introduce the breaking changes.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Given that the monthly Pandas release cycle, it's also impossible to update this documentation at every month.

Screenshot 2023-12-02 at 1 17 43 PM

File: python/docs/source/reference/pyspark.pandas/index.rst [hunk 2]
```diff
@@ -23,7 +23,7 @@ Pandas API on Spark
...
+   pandas API on Spark follows the API specifications of `Pandas 2.1.3 <https://pandas.pydata.org/pandas-docs/version/2.1.3/index.html>`_.
```
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply python/docs/source/reference/pyspark.pandas/index.rst:26]: +1, this one looks reasonable.

**Review by dongjoon-hyun — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**